### DISCUSSION TITLE
Get Instances from JSON Array Inside Object?

### DISCUSSION THREAD

**Discussion by SubutaDan** [flutterdata/flutter_data#274] [General]
I am trying to retrieve and display instances of Koko (each identified by kokoId).  The API method that retrieves the list of Kokos encloses the list in an object that contains a named array (the name is "items") that contains the Kokos:
```
{
 "items": [
  {
   "kokoId": "63ebb313-251d-40ad-bc0e-9693489b6a85",
   "kokoName": "klambake",
   "latitude": 457.0,
   "longitude": 22.0,
   "tags": [
    "d",
    "e",
    "f"
   ],
   "media": [
    {
     "kokoMedia_id": 9,
     "createdOn": "2020-11-30T09:49:58.515",
     "fileType": ".wav",
     "displayName": "klerf"
    },
    {
     "kokoMedia_id": 10,
     "createdOn": "2020-11-30T09:49:58.515",
     "fileType": ".jpg",
     "displayName": "smerf"
    },
    {
     "kokoMedia_id": 11,
     "createdOn": "2020-11-30T09:49:58.515",
     "fileType": ".jpeg",
     "displayName": "derf"
    }
   ],
   "createdOn": "2020-12-03T09:49:58.515",
   "description": "yee haw!",
   "locationProvenance": "LAST_KNOWN",
   "version": 1
  },
  {
   "kokoId": "0b1f17e8-8af1-4144-872f-3bc3fae691ad",
   "kokoName": "Oh dot of mine.",
   "latitude": 456.0,
. . . 
```
This form of results seems to be dictated by the API framework I am using (Google Cloud Endpoints), and I don't think I can change it.

I contrast this with the way Tasks are returned as an array of objects in the tutorial, which I think is the form expected by Flutter Data:
```
[
  {
    "id": 1,
    "title": "Laundry 🧺",
    "completed": false,
    "userId": 1
  },
  {
    "id": 2,
    "title": "Groceries 🛒",
    "completed": true,
    "userId": 1
  },
  {
    "id": 3,
    "title": "Reservation at Malloys",
    "completed": true,
    "userId": 1
  },
. . . 
```
Is there a way to get Flutter Data to access the Kokos as if they had been provided in the expected form, similar to the way it works with the Tasks in the tutorial?

Thanks very much.

-Dan

**Comment by frank06** [comment]
Yes definitely, you need to override the serializer. There are examples in the issue and you can also check out the JSON API adapter (it's another repo under flutterdata)
